test/ClangImporter/print-module-map-paths.swift

```diff
@@ -0,0 +1,38 @@
+// RUN: %empty-directory(%t)
+// RUN: mkdir -p %t/resources/linux/armv7 %t/sdk/usr/include %t/sdk/usr/lib/swift/linux/armv7
+// RUN: touch %t/sdk/usr/include/{inttypes,stdint,unistd}.h
+
+// RUN: touch %t/resources/linux/armv7/{SwiftGlibc.h,glibc.modulemap}
+// RUN: touch %t/sdk/usr/lib/swift/linux/armv7/{SwiftGlibc.h,glibc.modulemap}
+// RUN: %swift %s -typecheck -parse-stdlib -dump-clang-diagnostics -target armv7-unknown-linux-gnueabihf -sdk %t/sdk -resource-dir %t/resources 2>&1 | %FileCheck -check-prefix=CHECK-LINUX %s
+
+// RUN: cp %S/../../stdlib/public/Cxx/{cxxshim/libcxxshim.modulemap,libstdcxx/libstdcxx.h,libstdcxx/libstdcxx.modulemap} %t/resources/linux
+// RUN: if [ %target-os == "linux-gnu" ]; then %swift %s -typecheck -parse-stdlib -dump-clang-diagnostics -resource-dir %t/resources -cxx-interoperability-mode=default 2>&1 | %FileCheck -check-prefix=CHECK-LINUX-CXX %s; fi
```

Comment:
I tried to make this linux C++ Interop test run natively on other platforms, including mocking up some C++ headers, but [it apparently requires some separate gcc installation also](https://github.com/llvm/llvm-project/blob/8d17824890f668cca654a6b821d642bd3192dc81/clang/lib/Driver/ToolChains/Gnu.cpp#L3422), and I didn't want to mock that too. Simply passing in include paths with `-Xcc` without that gcc directory, which is actually unused for includes but seemingly required, wouldn't work, no matter what I tried.

So I have this test only run on linux with `%target-os` and check that it's correctly redirecting these module maps to the system default `-sdk /` instead. Technically, it requires that the compiler is able to access the C++ headers, so it really requires the host OS to be linux, but since nobody is likely running the linux tests on other platforms, it should work the same. If wanted, I can split it off into a separate linux-only test file instead.